bittorrent/sanitize.go

```diff
@@ -0,0 +1,50 @@
+package bittorrent
+
+import (
+	"errors"
+	"net"
+)
+
+// ErrInvalidIP indicates an invalid IP for an Announce.
+var ErrInvalidIP = errors.New("invalid IP")
+
+// RequestSanitizer is used to replace unreasonable values in requests parsed
+// from a frontend into sane values.
+type RequestSanitizer struct {
+	MaxNumWant          uint32 `yaml:"max_numwant"`
+	DefaultNumWant      uint32 `yaml:"default_numwant"`
+	MaxScrapeInfoHashes uint32 `yaml:"max_scrape_infohashes"`
+}
+
+// SanitizeAnnounce enforces a max and default NumWant and coerces the a peer's
+// IP address into the proper format.
+func (rs *RequestSanitizer) SanitizeAnnounce(r *AnnounceRequest) (*AnnounceRequest, error) {
+	if r.NumWant > rs.MaxNumWant {
+		r.NumWant = rs.MaxNumWant
+	}
+
+	if r.NumWant == 0 {
```

Comment:
Should we introduce a sentinel value to indicate `num_want` was missing rather than actually set to zero?